src/views/CashlinkManage.vue

```diff
@@ -0,0 +1,342 @@
+                if (cashlink) {
+                    this.retrievedCashlink = cashlink;
+                    this.isTxSent = true;
+                    this.isManagementRequest = true;
+                }
+            }
+        }
+
+        if (!this.isTxSent) {
+            if (!NetworkClient.hasInstance()) {
+                NetworkClient.createInstance(Config.networkEndpoint);
+            }
+            const network = NetworkClient.Instance;
+            await network.init();
+
+            if (!this.keyguardResult || !this.keyguardRequest) {
+                this.$rpc.reject(new Error('Unexpected: No valid Cashlink;'));
+                return;
+            }
+            // if there was a funding transaction the cashlink is in the static store.
+            this.retrievedCashlink.networkClient = network;
+            network.on(NetworkClient.Events.API_READY,
+                () => this.status = 'Contacting seed nodes...');
+            network.on(NetworkClient.Events.CONSENSUS_SYNCING,
+                () => this.status = 'Syncing consensus...');
+            network.on(NetworkClient.Events.CONSENSUS_ESTABLISHED,
+                () => this.status = 'Sending transaction...');
+            network.on(NetworkClient.Events.TRANSACTION_PENDING,
+                () => this.status = 'Awaiting receipt confirmation...');
+
+            network.relayTransaction({
+                sender: new Nimiq.Address(this.keyguardRequest.sender).toUserFriendlyAddress(),
+                senderPubKey: this.keyguardResult.publicKey,
+                recipient:  new Nimiq.Address(this.keyguardRequest.recipient).toUserFriendlyAddress(),
+                value: Nimiq.Policy.lunasToCoins(this.keyguardRequest.value),
+                fee: Nimiq.Policy.lunasToCoins(this.keyguardRequest.fee),
+                validityStartHeight: this.keyguardRequest.validityStartHeight,
+                signature: this.keyguardResult.signature,
+                extraData: this.keyguardRequest.data,
+            });
+
+            network.on(NetworkClient.Events.TRANSACTION_RELAYED, async (txInfo: any) => {
+                await CashlinkStore.Instance.put(this.retrievedCashlink!);
+                this.isTxSent = true;
+                window.setTimeout(() => this.state = StatusScreen.State.SUCCESS, StatusScreen.SUCCESS_REDIRECT_DELAY);
+            });
+        }
+    }
+
+    private get title(): string {
+        return this.state === StatusScreen.State.LOADING ? 'Creating your Cashlink' : 'Cashlink created.';
+    }
+
+    private get link(): string {
+        return `${window.location.origin}/${RequestType.CASHLINK}/#${this.retrievedCashlink!.render()}`;
+    }
+
+    private close() {
+        this.$rpc.resolve({
+            address: this.retrievedCashlink!.address.toUserFriendlyAddress(),
+            message: this.retrievedCashlink!.message,
+            status: this.retrievedCashlink!.state,
+        } as PublicCashlink);
+    }
```

Comment:
[Optional]
can we detect a window close and then also resolve instead of rejecting?

Reply:
I will experiment with this on my own cashlink branch where it also becomes more important that we resolve instead of rejecting on window close.